### PR TITLE
Replaces per-node CP-ABE encryption with AES

### PR DESCRIPTION
**Pull request by marcelamelara**

### CODE REVIEW

File: privateSBOMExchange/src/petra/lib/crypto.py [hunk 1]
```diff
@@ -1,4 +1,12 @@
 from __future__ import annotations
+import hashlib
+import sys
+import os
+
+from Crypto.Cipher import AES
+from Crypto.Random import get_random_bytes
```
Comment (SantiagoTorres) [inline privateSBOMExchange/src/petra/lib/crypto.py:7]: we need to add pycrypto/pycryptodome to requirements.txt

File: privateSBOMExchange/tests/test_cpabe_AES.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+from petra.lib.util.config import Config
+
+import cpabe
+
+argparser = argparse.ArgumentParser()
+# TODO: add args for the config
+# TODO: handle defaults etc
+argparser.add_argument("-o", "--original-file", type=str, required=True, help="the file to which to write the original SBOM tree")
+argparser.add_argument("-r", "--redacted-file", type=str, required=True, help="the file to which to write the redacted SBOM tree")
+argparser.add_argument("-d", "--decrypted-file", type=str, required=True, help="the file to which to write the decrypted SBOM tree")
```
Comment (SantiagoTorres) [inline privateSBOMExchange/tests/test_cpabe_AES.py:17]: I'm not sure if `-o` is overwriting the source file?

Comment (SantiagoTorres) [reply privateSBOMExchange/tests/test_cpabe_AES.py:17]: oh, the flag was confusing for me. Nvm

File: privateSBOMExchange/src/petra/lib/models/__init__.py [hunk 3]
```diff
@@ -626,7 +678,11 @@ def visit_field_node(self, node: FieldNode):
         """
         if node.encrypted_data != NODE_PUBLIC:
             try:
-                node.decrypted_data = bytes(cpabe_decrypt(self.secret_key, node.encrypted_data))
+                node.decrypted_data = decrypt_data_AES(node.encrypted_data, self.__decrypted_aes_keys[node.policy])
+                #node.decrypted_data = bytes(cpabe_decrypt(self.secret_key, node.encrypted_data))
```
Comment (SantiagoTorres) [inline privateSBOMExchange/src/petra/lib/models/__init__.py:682]: do we need this line?

**Review by SantiagoTorres — APPROVED** [review]
I could merge as is, but we could fix these two nits as well

File: privateSBOMExchange/src/petra/lib/models/__init__.py (hunk 3 above)
Comment (marcelamelara) [reply privateSBOMExchange/src/petra/lib/models/__init__.py:682]: I had left these in here _just in case_ we wanted to be able to switch, but I'll remove these

File: privateSBOMExchange/tests/test_cpabe_AES.py (hunk 2 above)
Comment (marcelamelara) [reply privateSBOMExchange/tests/test_cpabe_AES.py:17]: I'm also happy to rename the flag to something like `-p` or `-u` for pre- or un-redacted.

**Pull request merged**